sphinx_gallery/gen_rst.py

```diff
@@ -581,6 +581,9 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
                     example_rst += code_output
                 else:
                     example_rst += code_output
+                    if code_output:
+                        # Add some vertical space after output
```

Comment:
code_output is always true, even if the script produces no images, nor prints anything(pointless for an executable example maybe) code_output = '\n\n\n\n\n'